inc/config.class.php

```diff
@@ -3612,4 +3618,35 @@ public static final function generateUuid($type) {
       self::setConfigurationValues('core', [$type . '_uuid' => $uuid]);
       return $uuid;
    }
+
+   /**
+    * Get a file system adapter from configuration
+    *
+    * @param string $directory Directory
+    *
+    * @return Filesystem
+    */
+   public static function getFs($directory): Filesystem {
+      global $CFG_GLPI;
+
+      $conf_key = 'fs_adapater_' . md5($directory);
+      if (!isset($CFG_GLPI[$conf_key])) {
+         $CFG_GLPI[$conf_key] = new LocalFilesystemAdapter(
```

Comment:
I am not sure that storing an instance of an object in CFG_GLPI is a good idea.

If we want something that can be used later to be configured from UI or CLI, we should use a JSON entry in CFG_GLPI, like
```json
{
  "adapter": "LocalFilesystemAdapter",
  "params": {}
}
```
`params` entry will be required for AWS adapters.

Then instanciate the filesystem (or retrieve it from local private variable used for caching). Instanciation depending on config would require a switch/case logic, as constructors parameters are not harmonized through different adapters.

Reply:
Well, all of that depends on the goal; I'm not going to deal with things that are not supported. AWS is the problem of a potential implementation; we provide local only in core. If something like a plugin or else provide another adapter instance, it can be used as is.